pkg/profiling/profiling.go

```diff
@@ -0,0 +1,149 @@
+package profiling
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"math"
+	"os"
+	"path"
+	"runtime"
+	"runtime/pprof"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/pkg/logging"
+)
+
+var (
+	log = logging.LoggerForModule()
+)
+
+const (
+	fifoDefaultMaxFileCount = 10
+	heapdumpSubfolderName   = "heapdump"
+)
+
+type fifoDir struct {
```

Comment:
Looks like this object+method could go into a separate source file (probably too small for whole a separate package though).